### PR TITLE
Add Review Comment Model/API

### PR DESCRIPTION
**Pull request by thenav56**
Added New API endpoint.
- `entries/(?P<entry_id>\d+)/review-comments` (GET|POST)
- `entries/(?P<entry_id>\d+)/review-comments/{comment.id}/` (PUT|PATCH|DELETE)

Related Issues:
- Entry Review: Create APIs for entry review for particular entry #680 
- Entry Review: Create abstract model for review/comment #678

### CODE REVIEW

**Review by k9845 — APPROVED** [review]

File: apps/quality_assurance/models.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                'created_by__profile__display_picture',
+            ).prefetch_related(
+                'comment_texts',
+                'mentioned_users',
+                'mentioned_users__profile',
+                'mentioned_users__profile__display_picture',
+            ).filter(
+                models.Q(entry__lead__project__members=user) |
+                models.Q(entry__lead__project__user_groups__members=user)
+            ).distinct()
```
Comment (annshress) [inline apps/quality_assurance/models.py:61]: Maybe this part should go into appropriate model, rather than in abstract model, what do you think?

Comment (thenav56) [reply apps/quality_assurance/models.py:61]: Right now this matches both entities (entry and assessment), so I am leaving this here. We can always overwrite this as required on the model.

**Review by annshress — APPROVED** [review]

**Review by annshress — APPROVED** [review]

File: apps/quality_assurance/serializers.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            if not approved_by_qs.exists():
+                raise serializers.ValidationError({'comment_type': 'Need to approve first'})
+            entry.approved_by.remove(current_user)
+        elif comment_type == CommentType.CONTROL:
+            if entry.verified:
+                raise serializers.ValidationError({'comment_type': 'Already verified/controlled'})
+            entry.verify(current_user)
+        elif comment_type == CommentType.UNCONTROL:
+            if not entry.verified:
+                raise serializers.ValidationError({'comment_type': 'Need to verified/controlled first'})
```
Comment (tnagorra) [inline apps/quality_assurance/serializers.py]:
```suggestion
                raise serializers.ValidationError({'comment_type': 'Need to be verified/controlled first'})
```

Comment (thenav56) [reply apps/quality_assurance/serializers.py]: Done. Thanks.

**Review by k9845 — APPROVED** [review]

**Pull request merged**